packages/components/select/src/useSelect.ts

```diff
@@ -598,6 +599,15 @@ export const useSelect = (props: ISelectProps, emit) => {
     const valueKey = props.valueKey
     let index = -1
     arr.some((item, i) => {
+      if (valueKey !== SelectProps.valueKey.default) {
+        if (get(item.value, valueKey) === get(value.value, valueKey)) {
+          index = i
+          return true
+        } else {
+          return false
+        }
+      }
+
       if (toRaw(get(item, valueKey)) === get(value, valueKey)) {
```

Comment:
The above modification seems unnecessary, just replace the `get` method.
```suggestion
      if (getValueKey(item) === getValueKey(value) {
```

Reply:
@btea Ah Ok, I didn't know we've already had that function. I changed my code to your suggestion. Let try to take a look again. Thanks